cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/types/TypeParser.java

```diff
@@ -216,14 +175,10 @@ protected static boolean isQualifierSpecifier(String qualifier) {
    * @return true, if it is part of an elaborated type. false, otherwise
    */
   public static boolean isElaboratedTypeSpecifier(String specifier) {
-    if (getLanguage() instanceof CLanguage || getLanguage() instanceof CPPLanguage) {
-      return specifier.equals(ELABORATED_TYPE_CLASS)
-          || specifier.equals(ELABORATED_TYPE_STRUCT)
-          || specifier.equals(ELABORATED_TYPE_UNION)
-          || specifier.equals(ELABORATED_TYPE_ENUM);
-    }
-
-    return false;
+    return getLanguage() instanceof HasElaboratedTypeSpecifier
```

Comment:
Can we add language as an additional parameter here? I want to get rid of the `getLanguage` calls in the singleton type manager.